sentinelhub/config.py

```diff
@@ -48,156 +157,42 @@ class SHConfig:
-                raise ValueError("Value of config parameter 'max_opensearch_records_per_query' must be at most 500")
-
-            # The following enables that url parameters can be written with or without / at the end
-            for param, value in config.items():
-                if isinstance(value, str) and value.startswith("http"):
-                    config[param] = value.rstrip("/")
-
-            return config
-
-        def get_config_file(self) -> str:
-            """Checks if configuration file exists and returns its file path.
-            If file doesn't exist it creates a default configurations file.
-
-            :return: location of configuration file
-            """
-            config_file = os.path.join(os.path.dirname(__file__), "config.json")
-
-            if not os.path.isfile(config_file):
-                with open(config_file, "w") as cfg_file:
-                    json.dump(self.CONFIG_PARAMS, cfg_file, indent=2)
-
-            return config_file
-
-        def load_configuration(self) -> None:
-            """Method reads and loads the configuration file."""
-            with open(self.get_config_file(), "r") as cfg_file:
-                config = json.load(cfg_file)
-
-            config = self._parse_configuration(config)
-
-            for prop in config:
-                if prop in self.CONFIG_PARAMS:
-                    setattr(self, prop, config[prop])
-
-        def get_config(self) -> ConfigDict:
-            """Returns a dictionary with configuration parameters
-
-            :return: A dictionary
-            """
-            config = {item: getattr(self, item) for item in self.CONFIG_PARAMS}
-            for item, default_value in self.CONFIG_PARAMS.items():
-                if config[item] is None:
-                    config[item] = default_value
-            return config
-
-        def save_configuration(self) -> None:
-            """Method saves changed parameter values to the configuration file."""
-            config = self.get_config()
-
-            self._parse_configuration(config)
-
-            with open(self.get_config_file(), "w") as cfg_file:
-                json.dump(config, cfg_file, indent=2)
-
-    _instance: Optional[_SHConfig] = None
-
-    instance_id: str
-    sh_client_id: str
-    sh_client_secret: str
-    sh_base_url: str
-    sh_auth_base_url: str
-    geopedia_wms_url: str
-    geopedia_rest_url: str
-    aws_access_key_id: str
-    aws_secret_access_key: str
-    aws_session_token: str
-    aws_metadata_url: str
-    aws_s3_l1c_bucket: str
-    aws_s3_l2a_bucket: str
-    opensearch_url: str
-    max_wfs_records_per_query: int
-    max_opensearch_records_per_query: int  # pylint: disable=invalid-name
-    max_download_attempts: int
-    download_sleep_time: float
-    download_timeout_seconds: float
-    number_of_download_processes: int
-
-    def __init__(self, hide_credentials: bool = False):
+    _instance: Optional["SHConfig"] = None
```

Comment:
One option is to use _BaseSHConfig here, since it literally is an unloaded SHConfig

Reply:
switched over to Dict. It is currently not any safer than using SHConfig, but maybe it's better for future development

Reply:
Ok, I like the improvements. :+1: 

One last question - do we now even need to separate between `_BaseSHConfig` and `SHConfig`? Because in all case where you initialize `default = _BaseSHConfig()` you could now simply do `default = SHConfig(use_defaults=True)` and I don't see any danger if inifinite recursion. So I'm trying to see if there is an actual reason to have this separation to 2 different classes.

Reply:
Yes, I think it would better to make it a single class now. Otherwise, it might confuse people that would look at the code or the documentation of 2 classes.